tests/o3/cartesian_spherical_harmonics_test.py

```diff
@@ -159,3 +160,15 @@ def build_module(l, normalize, normalization):
 
     sp_pt2 = torch.compile(prepare(build_module)(l, normalize, normalization), fullgraph=True)
     assert torch.allclose(sp_pt2(xyz), sp(xyz))
+
+
+def test_internal_jit_flag():
+    l = o3.Irreps("0e + 1o + 3o")
+    sp = o3.SphericalHarmonics(l, normalization="integral", normalize=True)
+    # Turning off the JIT in _spherical_harmonics to enable torch.compile.
+    # Note that this is a less invasive way then turning off jit_script_fx globally
+    set_optimization_defaults(jit_script_fx=False)
```

Comment:
Need to do a context manager with a `try: finally` for this so it doesn't apply to all tests after.

Actually would not be bad to offer `set_optimization_defaults` as a context manager as well in `e3nn`